test/api/register.account.test.js

```diff
@@ -20,13 +20,18 @@ describe('API register account POW', () => {
 
 	describe('register account', () => {
 		it('register account', async () => {
-			const result = await echo.api.registerAccount(
-				'kokoko'+ Date.now(),
-				'ECHODvHDsAfk2M8LhYcxLZTbrNJRWT3UH5zxdaWimWc6uZkH',
-				'ECHODvHDsAfk2M8LhYcxLZTbrNJRWT3UH5zxdaWimWc6uZkH',
-				() => console.log('was broadcasted'),
-			)
-			ok(Array.isArray(result));
+			const result = new Promise(async (resolve) => {
+				return resolve(echo.api.registerAccount(
+						'kokoko'+ Date.now(),
+						'ECHODvHDsAfk2M8LhYcxLZTbrNJRWT3UH5zxdaWimWc6uZkH',
+						'ECHODvHDsAfk2M8LhYcxLZTbrNJRWT3UH5zxdaWimWc6uZkH',
+						() => console.log('was broadcasted'),
+					));
+			});
+			setTimeout(async () => await echo.api.getAccountCount(), 100);
+			ok(!Array.isArray(result));
+			const registrationResult = await result;
+			ok(Array.isArray(registrationResult));
```

Comment:
This check is not validate, that promise is resolved strictly after `getAccountCount`
Add variable
```ts
let isResolved = false;
```
Add in broadcasting callback
```ts
isResolved = true;
```
After `getAccountCount` check, that this variable is equal to `false`
Then await variable `result` (and rename it from `result` to `promise`, cause this is not a result).
And then again check `isResolved` to equal to `true`.